docs/metadata.md

```diff
@@ -0,0 +1,151 @@
+
+# Enable Collection
+
+By default metadata collection is turned off. 
+To enable collection:
+
+- Open the benchmark yaml that you will be running (for example byowl below)
+```
+apiVersion: ripsaw.cloudbulldozer.io/v1alpha1
+kind: Benchmark
+metadata:
+  name: byowl-benchmark
+  namespace: my-ripsaw
+spec:
+  workload:
+    name: byowl
+    args:
+      image: "quay.io/jtaleric/uperf:testing"
+      clients: 1
+      commands: "echo Test"
+```
+
+- Add ```metadata_collection: true``` to the spec section of the yaml
+```
+apiVersion: ripsaw.cloudbulldozer.io/v1alpha1
+kind: Benchmark
+metadata:
+  name: byowl-benchmark
+  namespace: my-ripsaw
+spec:
+  metadata_collection: true
+  workload:
+    name: byowl
+    args:
+      image: "quay.io/jtaleric/uperf:testing"
+      clients: 1
+      commands: "echo Test"
+```
+
+The metadata collection will now be enabled however as there is no ElasticSearch information
+defined it will fail.
+
+- Add the ElasticSearch server and Port information
+```
+apiVersion: ripsaw.cloudbulldozer.io/v1alpha1
+kind: Benchmark
+metadata:
+  name: byowl-benchmark
+  namespace: my-ripsaw
+spec:
+  es_server: "my.elastic.server.foo"
+  es_port: 9200
+  metadata_collection: true
+  workload:
+    name: byowl
+    args:
+      image: "quay.io/jtaleric/uperf:testing"
+      clients: 1
+      commands: "echo Test"
+```
+
+The metadata collection will now run as described previously and continue to
+the defined benchmark.
+
+# Additional Options
+
+There are a few additional options that can be enabled to enahnce the amount
+of data collected.
+
+## Privleged Pods
+
+By default, pods are run in an unprivledged state. While this makes permissions
+a lesser issue, it does limit the amount of data collected. For example, dmidecode
+requires privledges to read the memory information and generate the data.
+
+To enable privleged pods set:
+```
+metadata_privledged: true
+```
+
+In the spec section of the benchmark yaml as outlined for the other variables.
+
+## Kubeconfig
+
+There are multiple kubernetes (k8s) based modules that are run in stockpile that
+will not be run in the default configuration. To enable this collection you will
+need to provide a base64 encoded copy of you kubeconfig as the metadata_kubeconfig variable.
+
+To generate the data for this variable take your kubeconfig file (ex: /home/foo/kubeconfig)
+and run the following:
+```
+$ base64 /home/foo/kubeconfig | tr -d '\n'
+```
+
+This will provide a large string of data that you will need to add to the benchmark yaml.
+A completed yaml would like similiar to:
+```
+apiVersion: ripsaw.cloudbulldozer.io/v1alpha1
+kind: Benchmark
+metadata:
+  name: byowl-benchmark
+  namespace: my-ripsaw
+spec:
+  es_server: "my.elastic.server.foo"
+  es_port: 9200
+  metadata_collection: true
+  metadata_privledged: true
+  metadata_kubeconfig: SDIOUFGYASD9H87OIJFDS098FDISDJF098SIADJFOIJ309F.......
```

Comment:
Following up on this. The original attempt did not flesh out since it would require ripsaw to create an account with greater privileges than it has. So I documented the steps to create the service account and it can be passed via the yaml file.